inference-engine/src/mkldnn_plugin/nodes/mkldnn_rnn.cpp

```diff
@@ -263,53 +295,65 @@ void MKLDNNRNN::fillSeqDesc() {
         out_states_d[i - 1] = {S_4D_shape, memory::data_type::f32, memory::format_tag::ldnc};
     }
 
-    auto blobs = rnnLayer->blobs;
-    Blob::Ptr weights, bias;
-    if (blobs.find("weights") != blobs.end()) weights = blobs["weights"];
-    if (blobs.find("biases") != blobs.end()) bias = blobs["biases"];
-
-    if (!weights)
-        IE_THROW() << "RNN Layer. Weights do not present.";
-
-    if (weights->size() != G*SC*(SC+DC))
-        IE_THROW() << "RNN Layer. Weights size is not correct. Expected size:" << G*SC*(SC+DC);
+    in_states_d[0]  = {S_4D_shape, dataType, memory::format_tag::ldnc};
+    out_states_d[0] = {S_4D_shape, dataType, memory::format_tag::ldnc};
 
```

Comment:
I agree this code is a bit confusing, i was trying to keep the author idea.
We can have X inputs, the first input is "the real" input of the node.
in_states_d can contain:
- second input (hidden state)
- third input (cell state) (if present)
So in_states_d[0] is hidden state and in_states_d[1] (if present) is cell state.
Maybe it would be better to align this peace of code with Cell version, since it looks more clean without the loop